src/meta/src/manager/catalog/fragment.rs

```diff
@@ -1107,8 +1110,17 @@ impl FragmentManager {
             let table_fragments = map
                 .get(&table_id)
                 .with_context(|| format!("table_fragment not exist: id={}", table_id))?;
-            if let Some(fragment) = table_fragments.mview_fragment() {
-                fragments.insert(table_id, fragment);
+            match table_job_type.as_ref() {
+                Some(TableJobType::SharedCdcSource) => {
+                    if let Some(fragment) = table_fragments.source_fragment() {
+                        fragments.insert(table_id, fragment);
+                    }
+                }
+                _ => {
```

Comment:
Enumerate all cases as much as possible

```suggestion
                Some(TableJobType::Normal) => {...}
                _ => unreachable!()
```